### PR TITLE
Added Audience field in duckv1.Destination

### LINKED ISSUE #2796: Add `Audience` field in `duckv1.Destination`
As the [Eventing OIDC feature track](https://docs.google.com/document/d/1e7UgNTkL0Br5Da09Rg2ieVmhKJo4VXuBj-mHT9NCujY/edit#heading=h.oav9ejx0j1pc) describes, the [`Destination`](https://github.com/knative/pkg/blob/main/apis/duck/v1/destination.go) type allows to reference Addressable resources, raw Kubernetes services or absolute URLs. As for Addressables, it is necessary to additionally specify the required audience for those services.

Therefor the `Destination` type needs to allow to additionally specify the destinations `Audience`:

```
type Destination struct {

	...

	// Audience is the OIDC audience. 
	// This need only be set, if the target is not an Addressable 
	// and thus the Audience can't be received from the Addressable itself.
	// In case the Addressable specifies an Audience too, the Destinations
	// Audience takes preference.
	// +optional
	Audience *string `json:"audience,omitempty"`
}
```

**Additional context (optional)**

/good-first-issue

### PR DESCRIPTION
**Pull request by abhijeetgauravm**
# Changes

- Added Audience field in duckv1.Destination

/kind <kind>

Fixes #2796 

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Comment by creydr** [comment]
Hello @abhijeetgauravm,
thanks for your PR. Most of the knative repositories have a `hack` folder with some scripts (e.g. to update dependencies or to update generated code). You need to run `hack/update-codegen.sh` to generate/update the `DeepCopy` functions to include the new field.

**Comment by abhijeetgauravm** [comment]
> Hello @abhijeetgauravm, thanks for your PR. Most of the knative repositories have a [`hack` ](https://github.com/knative/pkg/tree/main/hack)folder with some scripts (e.g. to update dependencies or to update generated code). You need to run [`hack/update-codegen.sh` ](https://github.com/knative/pkg/blob/main/hack/update-codegen.sh) to generate/update the `DeepCopy` functions to include the new field.

@creydr done with the required changes!

File: apis/duck/v1/destination.go
Comment (creydr) [inline apis/duck/v1/destination.go:1]: @abhijeetgauravm can you fix these code style issues?

Comment (abhijeetgauravm) [reply apis/duck/v1/destination.go:1]: hey @creydr do i suppose to remove the empty line here too?

**Review by creydr — COMMENTED** [review]
Thanks for your contribution @abhijeetgauravm 
/lgtm

/assign @pierDipi

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**